src/react-components/src/__tests__/components/Link-test.tsx

```diff
@@ -0,0 +1,90 @@
+import { expect } from "chai";
+import { stub } from "sinon";
+import * as Enzyme from "enzyme";
+import * as React from "react";
+import * as Mocha from "mocha";
+
+import Link from "../../components/Link/Link";
+import { LinkTypes, LinkIconPositions } from "../../components/Link/LinkTypes";
+import Icon from "../../components/Icons/Icon";
+import { iconRotationTypes } from "../../components/Icons/IconTypes";
+
+describe("Link", () => {
+    let wrapper: Enzyme.ReactWrapper<{}, {}>;
+
+    it("Can pass in icon, text as child and url as props", () => {
+        wrapper = Enzyme.mount(
+            <Link href="#passed-in-link" linkType={LinkTypes.Action}>
+                <Icon
+                    name="download"
+                    blockName="more-link"
+                    modifiers={["left"]}
+                    decorative={true}
+                    iconRotation={iconRotationTypes.rotate0}
+                ></Icon>
+                Download
+            </Link>
+        );
+        expect(wrapper.find("a")).to.have.lengthOf(1);
+        expect(wrapper.find("Icon")).to.have.lengthOf(1);
+    });
+
+    it("Can pass a link with <a> tag", () => {
+        wrapper = Enzyme.mount(
+            <Link>
+                <a href="#test">Test</a>
+            </Link>
+        );
+        expect(wrapper.find("a")).to.have.lengthOf(1);
+    });
+
+    it("Can pass a link with <a> tag and an icon", () => {
+        wrapper = Enzyme.mount(
+            <Link>
+                <a href="#test">Test</a>
+                <Icon
+                    name="download"
+                    blockName="more-link"
+                    modifiers={["left"]}
+                    decorative={true}
+                    iconRotation={iconRotationTypes.rotate0}
+                ></Icon>
+            </Link>
+        );
+        expect(wrapper.find("a")).to.have.lengthOf(1);
+        expect(wrapper.find("Icon")).to.have.lengthOf(1);
+    });
+
+    it("Can generate a back link", () => {
+        wrapper = Enzyme.mount(
+            <Link href="#passed-in-link" linkType={LinkTypes.Backwards}>
+                content
+            </Link>
+        );
+        expect(
+            wrapper.find(".more-link__icon").hasClass("more-link__icon--left")
+        ).to.equal(true);
+    });
+
+    it("Can pass in right icon", () => {
```

Comment:
An icon isn't being passed here, I think this test is better called `LinkType Forwards generates right icon`

Reply:
There should be a different passed in link test.

Reply:
A different passed in link test?

Reply:
Hmm ... actually, I think that's not necessary since the functionality is shown in stories and it's really only testing react.